### DISCUSSION TITLE
Matek ELRS PWM + P1-Gyro initialization issue

### DISCUSSION THREAD

**Discussion by Blastaar-BR** [ExpressLRS/ExpressLRS#1657] [Q&A]
Hi,

I got an issue with P1-Gyro and Matek ELRS PWM 5 ch rx FW V 2.5.

1st step I used the P1-Gyro with a Radiomaster R88 receiver and it worked ok. I simply replaced the R88 by the Matek ELRS and a weird thing happened:

When I plug on the plane battery the ELRS rx binds ok with Radiomaster T16s with Naminmo Flash Oled tx V.2.5,  however the P1 seems to be frozen, no LED blinks, nothing happen (no response to any channel changes in the rc - seems that it does not initialize).

Tried to sort out and realized that I need to first turn on the ELRS rx and after it binds to the rc, I turn on the P1-Gyro and it "recognizes" the rx and start working.

If I turn on both rx and P1 together P1 gets frozen.

I replaced the P1 with two other spare units, and behaviour is the same. Looks like the P1 does not like the Matek ELRS PWM, once it works fine with R88 rx.

Any way to reproduce it and find a solution?

I appreciate any help!

**Comment by commanderguy3001** [comment]
PWM in general still isn't perfect with ELRS yet, but a couple things you can try that helped in other places with various pwm issues:
- set the packet rate from the expresslrs lua to 500hz, and try if that helps
- if you wanna get creative: you could try and flash the master branch to the receiver and transmitter, and check if that changes anything

otherwise I don't have a ready-made solution, but probably some of the devs could help out in that case.

**Reply by Blastaar-BR** [reply]
Hi, 

I tried to increase the packet rate to 500 hz but it did not help. The issue remains.

Tried to connect the P1-Gyro with different brands and types of PWM rx (Radiomaster R88, R168, Jumper D8) and all work fine.

The P1-Gyro is missing something when interfacing with the ELRS PWM rx during the power up.

If someone could take a look on that would be really fantastic once the ELRS features are amazing for planes too.

Anyway, thank you so much for the help so far. :)

Regarding the FW to be flashed: I am not sure which FW should I flash now as a trial when you say master branch ... are you suggesting trying the V3.0 pre-release (I saw improvements in PWM area)?
